wallet/src/ui/app/redux/slices/sui-objects/index.ts

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { getObjectExistsResponse } from '@mysten/sui.js';
+import {
+    createAsyncThunk,
+    createEntityAdapter,
+    createSlice,
+} from '@reduxjs/toolkit';
+
+import type { SuiObject } from '@mysten/sui.js';
+import type { AppThunkConfig } from '_store/thunk-extras';
+
+const objectsAdapter = createEntityAdapter<SuiObject>({
+    selectId: ({ reference }) => reference.objectId,
+});
+
+export const fetchAllOwnedObjects = createAsyncThunk<
+    SuiObject[],
+    void,
+    AppThunkConfig
+>('sui-objects/fetch-all', async (_, { getState, extra: { api } }) => {
+    const address = getState().account.address;
+    const allSuiObjects: SuiObject[] = [];
+    if (address) {
+        const allObjectRefs = await api.instance.getOwnedObjectRefs(
+            `${address}`
+        );
+        const objectIDs = allObjectRefs.map((anObj) => anObj.objectId);
+        const allObjRes = await api.instance.getObjectInfoBatch(objectIDs);
+        for (const objRes of allObjRes) {
+            const suiObj = getObjectExistsResponse(objRes);
+            if (suiObj) {
+                allSuiObjects.push(suiObj);
```

Comment:
```suggestion
                allSuiObjects.push(objRes);
```

I would recommend using `objRes` here so that you can leverage the the Coin utility class in the other file: https://github.com/MystenLabs/fastnft/blob/16c817330275a97fac67534a104f8a363571efea/sdk/typescript/src/types/framework.ts#L14

Feel free to modify https://github.com/MystenLabs/fastnft/blob/16c817330275a97fac67534a104f8a363571efea/sdk/typescript/src/types/framework.ts#L14 to add a `getSymbol` function if needed

Reply:
It doesn't really make sense to store the whole response, I moved some of the functionality to a utility class already but probably will move it to sdk and reuse that.